src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1421,92 +1424,152 @@ bool OutputStateMachineEngine::s_ParseColorSpec(const std::wstring_view string,
+        {
+            return false;
+        }
+
+        foundRGB = true;
+        rgbHexDigitCount = ((stringSize - 3) / 3) - 1;
+    }
+
+    // Try the sharp sign format.
+    curr--;
+    if (*curr++ == L'#')
+    {
+        // We can have one of the following formats:
+        // 4 "#hhh"
+        // 7 "#hhhhhh"
+        // 10 "#hhhhhhhhh"
+        // 13 "#hhhhhhhhhhhh"
+        // Any other cases will be invalid.
+        if (!(stringSize == 4 || stringSize == 7 || stringSize == 10 || stringSize == 13))
+        {
+            return false;
+        }
+
+        isSharpSignFormat = true;
         foundRGB = true;
+        rgbHexDigitCount = (stringSize - 1) / 3;
     }
 
     if (foundRGB)
     {
-        // Colorspecs are up to hh/hh/hh, for 1-2 h's
+        // Colorspecs are up to hhhh/hhhh/hhhh, for 1-4 h's
         for (size_t component = 0; component < 3; component++)
         {
             bool foundColor = false;
-            auto& value = til::at(colorValues, component);
-            for (size_t i = 0; i < 3; i++)
+            auto& parameterValue = til::at(parameterValues, component);
+            for (size_t i = 0; i < rgbHexDigitCount; i++)
             {
                 const wchar_t wch = *curr++;
 
                 if (_isHexNumber(wch))
                 {
-                    value *= 16;
+                    parameterValue *= 16;
                     unsigned int intVal = 0;
                     if (s_HexToUint(wch, intVal))
                     {
-                        value += intVal;
+                        parameterValue += intVal;
                     }
                     else
                     {
                         // Encountered something weird oh no
                         foundColor = false;
                         break;
                     }
-                    // If we're on the blue component, we're not going to see a /.
-                    // Break out once we hit the end.
-                    if (component == 2 && curr >= string.cend())
-                    {
-                        foundValidColorSpec = true;
-                        break;
-                    }
                 }
-                else if (wch == L'/')
+            }
+
+            if (isSharpSignFormat)
+            {
+                // Successfully parsed this component. Start the next one.
+                foundColor = true;
+            }
+            else
+            {
+                if (curr < string.cend() && *curr == L'/')
                 {
-                    // Break this component, and start the next one.
+                    // Same as above but need to skip the delimiter.
+                    curr++;
                     foundColor = true;
-                    break;
                 }
                 else
                 {
-                    // Encountered something weird oh no
+                    // Invalid delimiter. Treat it as an error.

```

Comment:
I wonder if this is a bit too strict. Technically this is the same logic as before.